### PR TITLE
gitlab: Rewrite plugin to use new API client.

### LINKED ISSUE #188: Add access request notes to Gitlab integration
When teleport-gitlab creates a new gitlab issue, add the requests note to the issue body.

### LINKED ISSUE #209: Update Gitlab, Pagerduty and JIRA
* [ ] Use new API
* [ ] Switch to the same UX
Prep for release of 6.2.0

### PR DESCRIPTION
**Pull request by marshall-lee**
- Use new `teleport/api` client.
- Add comments to the issue when new access review is submitted.
- Automatically change issue status once request is approved/denied.
- Manage plugin data state safely using compare-and-swap.
- Migrate tests to testify.
- Also update slack, mattermost and pagerduty plugins with a new `addr` config setting instead of deprecated `auth_server`.
- Add exponential backoff to `modifyPluginData` loop in slack, mattermost and pagerduty plugins.

Closes #188.
Closes #209.

### CODE REVIEW

**Comment by tcsc** [comment]
Are there any UX-impacting changes here for @klizhentas to review, or should we proceed with buddy PR?

File: access/gitlab/README.md [hunk 1]
```diff
@@ -1,31 +1,89 @@
...
 [teleport]
-auth_server = "example.com:3025"                         # Teleport Auth Server GRPC API address
-client_key = "/var/lib/teleport/plugins/gitlab/auth.key" # Teleport GRPC client secret key
-client_crt = "/var/lib/teleport/plugins/gitlab/auth.crt" # Teleport GRPC client certificate
-root_cas = "/var/lib/teleport/plugins/gitlab/auth.cas"   # Teleport cluster CA certs
+# Teleport Auth/Proxy Server address.
+#
+# Should be port 3080 or 443 for proxy.
+# For Teleport Cloud, should be in the form of "your-account.teleport.sh".
+auth_server = "example.com:3025"
```
Comment (klizhentas) [inline access/gitlab/README.md:82]: take a look at https://github.com/gravitational/teleport-plugins/tree/master/event-handler

the field `auth_server` is called `addr` there. Can you unify both field names? (e.g. support addr and deprecate auth_server by removing from the docs, but supporting it for transition)

**Review by klizhentas — COMMENTED** [review]
Have a quick comment on the config file name, otherwise LGTM

File: access/gitlab/README.md (hunk 1 above)
Comment (marshall-lee) [reply access/gitlab/README.md]: Done!

Slack, Mattermost and Pagerduty plugins are also updated. Jira plugin updated in #212

**Pull request closed without merging**